### DISCUSSION TITLE
Delegating to a validator

### DISCUSSION THREAD

**Discussion by roccomuso** [0xPolygon/polygon-edge#279] [Q&A] [answered]
## [How to delegate using SDK]

I'm not able to find any doc related to Staking.

Is there any documentation around that shows how to craft a Delegation tx?
Also any rpc that will return some info about the amount staked etc.?

**Comment by zivkovicmilos** [comment]
Hey @roccomuso,

Thank you for reaching out, and I apologize for the late reply.

We're currently in the process of adding docs that have information on our staking implementation.
They should be up soon, and I'll make sure to ping you here once they're live so you can take a look.

Our current staking implementation is located on the `ibft-pos-v3` branch, and is not considered stable for general use yet.

We don't have any kind of delegation mechanism currently, but plan to add it in the future.
Could you please tell us more about what you're trying to build with Polygon SDK, as well as what features you'd like to have (specifically in terms of staking)?

**Comment by roccomuso** [comment]
Thanks, I'll wait for the doc.

What I'm trying to achieve is just allow to craft a tx to "stake" and "unstake". That's it

**Comment by MagnusTran** [comment]
Waiting for this feature also.

**Comment by zivkovicmilos** [comment, marked as answer]
Hey @roccomuso and @MagnusTran,

We're working on integrating the PoA and PoS consensus into the official Polygon SDK releases, so users can choose which one they want seamlessly (PR #219)

We'll also be providing the docs along with this addition.
The current time frame looks ~2 weeks from now.

I'll ping you here once we merge out these features 👍

**Comment by lazartravica** [comment]
@MagnusTran @roccomuso Just for additional clarification, what @zivkovicmilos is mentioning is a consensus feature of Polygon-SDK, as a tool & a fully featured blockchain client for spinning up your own ethereum-compatible networks. 

We are not working on a feature to create stake/unstake transactions for the Polygon PoS mainnet from JavaScript or any other clientside/backend piece of software, but you can ping the folks at the Discord server for assistance if that's what you're looking for.

Best regards!
